File: Nx3CS/bin/test/src/nx3/xml/PartXML.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3.xml{
	public  class PartXML : global::haxe.lang.HxObject {
		static PartXML() {
			#line 19 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
			global::nx3.xml.PartXML.XPART = "part";
			global::nx3.xml.PartXML.XPART_TYPE = "type";
			global::nx3.xml.PartXML.XPART_CLEF = "clef";
			global::nx3.xml.PartXML.XPART_CLEFDISPLAY = "clefdisplay";
			global::nx3.xml.PartXML.XPART_KEY = "key";
			global::nx3.xml.PartXML.XPART_KEYDISPLAY = "keydisplay";
		}
		public    PartXML(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PartXML(){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				global::nx3.xml.PartXML.__hx_ctor_nx3_xml_PartXML(this);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_xml_PartXML(global::nx3.xml.PartXML __temp_me150){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public static  string XPART;
		
		public static  string XPART_TYPE;
		
		public static  string XPART_CLEF;
		
		public static  string XPART_CLEFDISPLAY;
		
		public static  string XPART_KEY;
		
		public static  string XPART_KEYDISPLAY;
		
		public static   global::Xml toXml(global::nx3.NPart part){
			unchecked {
				#line 28 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				global::Xml xml = global::Xml.createElement("part");
				#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
					#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					int _g = 0;
					#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::Array<object> _g1 = part.nvoices;
					#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					while (( _g < _g1.length )){
						#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
						global::nx3.NVoice voice = ((global::nx3.NVoice) (_g1[_g]) );
						#line 31 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
						 ++ _g;
						#line 33 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
						global::Xml voiceXml = global::nx3.xml.VoiceXML.toXml(voice);
						xml.addChild(voiceXml);
					}
					
				}
				
				#line 38 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
					#line 38 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::nx3.EPartType _g2 = part.type;
					#line 38 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					switch (global::Type.enumIndex(_g2)){
						case 0:
						{
							#line 40 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							{
							}
							
							#line 40 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
						
						default:
						{
							#line 43 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							xml.@set("type", global::Std.@string(part.type));
							#line 43 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 49 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				if (( part.clef != default(global::nx3.EClef) )) {
					#line 50 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::nx3.EClef _g3 = part.clef;
					#line 50 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					switch (global::Type.enumIndex(_g3)){
						case 0:
						{
							#line 52 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							{
							}
							
							#line 52 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
						
						default:
						{
							#line 55 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							xml.@set("clef", global::Std.@string(part.clef));
							#line 55 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 58 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				if (( part.key != default(global::nx3.EKey) )) {
					#line 59 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::nx3.EKey _g4 = part.key;
					#line 59 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					switch (global::Type.enumIndex(_g4)){
						case 6:
						{
							#line 61 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							{
							}
							
							#line 61 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
						
						default:
						{
							#line 64 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							xml.@set("key", global::Std.@string(part.key));
							#line 64 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 69 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
					#line 69 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::nx3.EDisplayALN _g5 = part.clefDisplay;
					#line 69 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					switch (global::Type.enumIndex(_g5)){
						case 1:
						{
							#line 71 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							{
							}
							
							#line 71 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
						
						default:
						{
							#line 74 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							xml.@set("clefdisplay", global::Std.@string(part.clefDisplay));
							#line 74 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 91 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				{
					#line 91 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					global::nx3.EDisplayALN _g6 = part.keyDisplay;
					#line 91 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					switch (global::Type.enumIndex(_g6)){
						case 1:
						{
							#line 93 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							{
							}
							
							#line 93 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
						
						default:
						{
							#line 96 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							xml.@set("keydisplay", global::Std.@string(part.keyDisplay));
							#line 96 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 99 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				return xml;
			}
			#line default
		}
		
		
		public static   global::nx3.NPart fromXmlStr(string xmlStr){
			unchecked {
				#line 104 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				global::Xml xml = global::Xml.parse(xmlStr).firstElement();
				#line 106 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				global::Array<object> voices = new global::Array<object>(new object[]{});
				{
					#line 107 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					object __temp_iterator177 = xml.elements();
					#line 107 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					while (global::haxe.lang.Runtime.toBool(global::haxe.lang.Runtime.callField(__temp_iterator177, "hasNext", 407283053, default(global::Array)))){
						#line 107 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
						global::Xml v = ((global::Xml) (global::haxe.lang.Runtime.callField(__temp_iterator177, "next", 1224901875, default(global::Array))) );
						#line 109 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
						global::nx3.NVoice voice = global::nx3.xml.VoiceXML.fromXmlStr(v.toString());
						voices.push(voice);
					}
					
				}
				
				#line 114 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				string typeStr = xml.@get("type");
				global::nx3.EPartType type = global::cx.EnumTools.createFromString<global::nx3.EPartType>(typeof(global::nx3.EPartType), typeStr);
				#line 118 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				string str = xml.@get("clef");
				global::nx3.EClef clef = default(global::nx3.EClef);
				if ( ! (string.Equals(str, default(string))) ) {
					#line 120 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					clef = global::cx.EnumTools.createFromString<global::nx3.EClef>(typeof(global::nx3.EClef), str);
				}
				
				#line 123 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				string clefDisplayStr = xml.@get("clefdisplay");
				global::nx3.EDisplayALN clefDisplay = default(global::nx3.EDisplayALN);
				#line 124 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				if (string.Equals(clefDisplayStr, default(string))) {
					#line 124 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					clefDisplay = global::nx3.EDisplayALN.Layout;
				}
				 else {
					#line 124 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					clefDisplay = global::cx.EnumTools.createFromString<global::nx3.EDisplayALN>(typeof(global::nx3.EDisplayALN), clefDisplayStr);
				}
				
				#line 127 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				string str1 = xml.@get("key");
				global::nx3.EKey key = default(global::nx3.EKey);
				if ( ! (string.Equals(str1, default(string))) ) {
					#line 129 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					key = global::cx.EnumTools.createFromString<global::nx3.EKey>(typeof(global::nx3.EKey), str1);
				}
				
				#line 133 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				string keyDisplayStr = xml.@get("keydisplay");
				global::nx3.EDisplayALN keyDisplay = default(global::nx3.EDisplayALN);
				#line 134 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				if (string.Equals(keyDisplayStr, default(string))) {
					#line 134 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					keyDisplay = global::nx3.EDisplayALN.Layout;
				}
				 else {
					#line 134 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
					keyDisplay = global::cx.EnumTools.createFromString<global::nx3.EDisplayALN>(typeof(global::nx3.EDisplayALN), keyDisplayStr);
				}
				
				#line 136 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				return new global::nx3.NPart(((global::Array<object>) (voices) ), ((global::nx3.EPartType) (type) ), ((global::nx3.EClef) (clef) ), ((global::nx3.EDisplayALN) (clefDisplay) ), ((global::nx3.EKey) (key) ), ((global::nx3.EDisplayALN) (keyDisplay) ));
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				return new global::nx3.xml.PartXML(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\xml\\PartXML.hx"
				return new global::nx3.xml.PartXML();
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/xml/PartXML.java
package nx3.xml;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PartXML extends haxe.lang.HxObject
{
	static 
	{
		nx3.xml.PartXML.XPART = "part";
		nx3.xml.PartXML.XPART_TYPE = "type";
		nx3.xml.PartXML.XPART_CLEF = "clef";
		nx3.xml.PartXML.XPART_CLEFDISPLAY = "clefdisplay";
		nx3.xml.PartXML.XPART_KEY = "key";
		nx3.xml.PartXML.XPART_KEYDISPLAY = "keydisplay";
	}
	public    PartXML(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PartXML()
	{
		nx3.xml.PartXML.__hx_ctor_nx3_xml_PartXML(this);
	}
	
	
	public static   void __hx_ctor_nx3_xml_PartXML(nx3.xml.PartXML __temp_me193)
	{
		{
		}
		
	}
	
	
	public static  java.lang.String XPART;
	
	public static  java.lang.String XPART_TYPE;
	
	public static  java.lang.String XPART_CLEF;
	
	public static  java.lang.String XPART_CLEFDISPLAY;
	
	public static  java.lang.String XPART_KEY;
	
	public static  java.lang.String XPART_KEYDISPLAY;
	
	public static   haxe.root.Xml toXml(nx3.NPart part)
	{
		haxe.root.Xml xml = haxe.root.Xml.createElement("part");
		{
			int _g = 0;
			haxe.root.Array<nx3.NVoice> _g1 = part.nvoices;
			while (( _g < _g1.length ))
			{
				nx3.NVoice voice = _g1.__get(_g);
				 ++ _g;
				haxe.root.Xml voiceXml = nx3.xml.VoiceXML.toXml(voice);
				xml.addChild(voiceXml);
			}
			
		}
		
		{
			nx3.EPartType _g2 = part.type;
			switch (haxe.root.Type.enumIndex(_g2))
			{
				case 0:
				{
					{
					}
					
					break;
				}
				
				
				default:
				{
					xml.set("type", haxe.root.Std.string(part.type));
					break;
				}
				
			}
			
		}
		
		if (( part.clef != null )) 
		{
			nx3.EClef _g3 = part.clef;
			switch (haxe.root.Type.enumIndex(_g3))
			{
				case 0:
				{
					{
					}
					
					break;
				}
				
				
				default:
				{
					xml.set("clef", haxe.root.Std.string(part.clef));
					break;
				}
				
			}
			
		}
		
		if (( part.key != null )) 
		{
			nx3.EKey _g4 = part.key;
			switch (haxe.root.Type.enumIndex(_g4))
			{
				case 6:
				{
					{
					}
					
					break;
				}
				
				
				default:
				{
					xml.set("key", haxe.root.Std.string(part.key));
					break;
				}
				
			}
			
		}
		
		{
			nx3.EDisplayALN _g5 = part.clefDisplay;
			switch (haxe.root.Type.enumIndex(_g5))
			{
				case 1:
				{
					{
					}
					
					break;
				}
				
				
				default:
				{
					xml.set("clefdisplay", haxe.root.Std.string(part.clefDisplay));
					break;
				}
				
			}
			
		}
		
		{
			nx3.EDisplayALN _g6 = part.keyDisplay;
			switch (haxe.root.Type.enumIndex(_g6))
			{
				case 1:
				{
					{
					}
					
					break;
				}
				
				
				default:
				{
					xml.set("keydisplay", haxe.root.Std.string(part.keyDisplay));
					break;
				}
				
			}
			
		}
		
		return xml;
	}
	
	
	public static   nx3.NPart fromXmlStr(java.lang.String xmlStr)
	{
		haxe.root.Xml xml = haxe.root.Xml.parse(xmlStr).firstElement();
		haxe.root.Array<nx3.NVoice> voices = new haxe.root.Array<nx3.NVoice>(new nx3.NVoice[]{});
		{
			java.lang.Object __temp_iterator245 = xml.elements();
			while (haxe.lang.Runtime.toBool(haxe.lang.Runtime.callField(__temp_iterator245, "hasNext", null)))
			{
				haxe.root.Xml v = ((haxe.root.Xml) (haxe.lang.Runtime.callField(__temp_iterator245, "next", null)) );
				nx3.NVoice voice = nx3.xml.VoiceXML.fromXmlStr(v.toString());
				voices.push(voice);
			}
			
		}
		
		java.lang.String typeStr = xml.get("type");
		nx3.EPartType type = cx.EnumTools.createFromString(nx3.EPartType.class, typeStr);
		java.lang.String str = xml.get("clef");
		nx3.EClef clef = null;
		if (( str != null )) 
		{
			clef = cx.EnumTools.createFromString(nx3.EClef.class, str);
		}
		
		java.lang.String clefDisplayStr = xml.get("clefdisplay");
		nx3.EDisplayALN clefDisplay = null;
		if (( clefDisplayStr == null )) 
		{
			clefDisplay = nx3.EDisplayALN.Layout;
		}
		 else 
		{
			clefDisplay = cx.EnumTools.createFromString(nx3.EDisplayALN.class, clefDisplayStr);
		}
		
		java.lang.String str1 = xml.get("key");
		nx3.EKey key = null;
		if (( str1 != null )) 
		{
			key = cx.EnumTools.createFromString(nx3.EKey.class, str1);
		}
		
		java.lang.String keyDisplayStr = xml.get("keydisplay");
		nx3.EDisplayALN keyDisplay = null;
		if (( keyDisplayStr == null )) 
		{
			keyDisplay = nx3.EDisplayALN.Layout;
		}
		 else 
		{
			keyDisplay = cx.EnumTools.createFromString(nx3.EDisplayALN.class, keyDisplayStr);
		}
		
		return new nx3.NPart(((haxe.root.Array<nx3.NVoice>) (voices) ), ((nx3.EPartType) (type) ), ((nx3.EClef) (clef) ), ((nx3.EDisplayALN) (clefDisplay) ), ((nx3.EKey) (key) ), ((nx3.EDisplayALN) (keyDisplay) ));
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.xml.PartXML(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.xml.PartXML();
	}
	
	
}


